File: app/src/main/java/br/com/jonathanzanella/myexpenses/bill/BillContract.java
package br.com.jonathanzanella.myexpenses.bill;

import android.content.Context;
import android.support.annotation.StringRes;

import org.joda.time.DateTime;

import br.com.jonathanzanella.myexpenses.validations.ValidationError;

interface BillContract {
	interface View {
		Context getContext();
		void setTitle(@StringRes int string);
		void setTitle(String string);
		void showBill(Bill bill);
	}

	interface EditView extends View {
		Bill fillBill(Bill bill);
		void finishView();
		void showError(ValidationError error);
		void onInitDateChanged(DateTime date);
		void onEndDateChanged(DateTime date);
	}
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/bill/BillContract.kt
package br.com.jonathanzanella.myexpenses.bill

import android.content.Context
import android.support.annotation.StringRes
import br.com.jonathanzanella.myexpenses.validations.ValidationError
import org.joda.time.DateTime

internal interface BillContract {
    interface View {
        val context: Context
        fun setTitle(@StringRes string: Int)
        fun setTitle(string: String)
        fun showBill(bill: Bill)
    }

    interface EditView : View {
        fun fillBill(bill: Bill): Bill
        fun finishView()
        fun showError(error: ValidationError)
        fun onInitDateChanged(date: DateTime)
        fun onEndDateChanged(date: DateTime)
    }
}
